actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -2706,6 +2706,8 @@ def fields_for_with_nested_attributes(association_name, association, options, bl
             association = @object.public_send(association_name)
           end
 
+          options[:public_id_column] = @object.nested_attributes_options.dig(association_name.to_sym, :public_id_column) || :id
```

Comment:
This may be abusing the options hash for this but it seemed wiser than adding an additional argument to `fields_for_nested_model`